frontend/src/routes/candidate/(protected)/questions/[questionId]/+page.svelte

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+  import {_} from 'svelte-i18n';
+  import {goto} from '$app/navigation';
+  import {page} from '$app/stores';
+  import {answerContext} from '$lib/utils/answerStore';
+  import {Question} from '$lib/components/questions';
+  import BasicPage from '$lib/templates/basicPage/BasicPage.svelte';
+  import {addAnswer, updateAnswer} from '$lib/api/candidate';
+  import {get} from 'svelte/store';
+
+  const answers = get(answerContext.answers);
+  const answerStore = answerContext.answers;
+
+  /**
+   * A small delay before moving to the next question.
+   * TODO: Make this a global variable used throughout the app.
+   */
+  const DELAY_M_MS = 350;
+
+  let currentQuestion: QuestionProps | undefined;
+  $: currentQuestion = $page.data.questions.find((q) => '' + q.id === '' + $page.params.questionId);
+
+  $: answer = $answerStore[$page.params.questionId];
+
+  // Store question id and answer value in a store
+  async function answerQuestion({detail}: CustomEvent) {
+    if (!answer) {
+      const response = await addAnswer(detail.id, detail.value);
+      const data = await response.json();
+      answers[$page.params.questionId] = {id: data.data.id, key: detail.value};
+    } else {
+      await updateAnswer(answer.id, detail.value);
+      answers[$page.params.questionId] = {
+        id: answer.id,
+        key: detail.value
+      };
+    }
+
+    answerContext.answers.set(answers);
+  }
+
+  // Skip to next question
+  // TODO: Later we might want to add an explicit note that this question was skipped
+  // TODO: This needs to take into account as well whether the question is already answered
+  function skipQuestion() {
+    gotoNextQuestion();
+  }
+
+  function gotoNextQuestion() {
+    if (!currentQuestion) {
+      return;
+    }
+    const currentIndex = $page.data.questions.indexOf(currentQuestion);
+    if (currentIndex < $page.data.questions.length - 1) {
+      const nextId = $page.data.questions[currentIndex + 1].id;
+      const currentUrl = $page.url.pathname.replace(/\/$/, '');
+      const nextQuestionUrl = `${currentUrl.substring(0, currentUrl.lastIndexOf('/'))}/${nextId}`;
```

Comment:
Same thing as previous comment, can we use some kind of helper from Svelte that allows referencing specific paths properly?